src/ComparedQueryable/NativeQueryable/EnumerableExecutor.cs

```diff
@@ -34,11 +35,17 @@ public EnumerableExecutor(Expression expression)
 
         internal T Execute()
         {
-            EnumerableRewriter rewriter = new EnumerableRewriter();
+            EnumerableRewriter rewriter = GetEnumerableRewriter();
             Expression body = rewriter.Visit(_expression);
             Expression<Func<T>> f = Expression.Lambda<Func<T>>(body, (IEnumerable<ParameterExpression>)null);
             Func<T> func = f.Compile();
             return func();
         }
+
+        internal virtual EnumerableRewriter GetEnumerableRewriter()

```

Comment:
Had to extend these classes to allow for the generic subclasses to be able to extend its functionality.